packages/cli/src/commands/target/list.ts

```diff
@@ -0,0 +1,136 @@
+import chalk from 'chalk';
+import ms from 'ms';
+import table from '../../util/output/table';
+import Client from '../../util/client';
+import { getCommandName } from '../../util/pkg-name';
+import type { ProjectLinked } from '@vercel-internals/types';
+import type { CustomEnvironment } from '../../util/target/types';
+
+export default async function list(
+  client: Client,
+  argv: any,
+  args: string[],
+  link: ProjectLinked
+) {
+  const { output } = client;
+  if (args.length !== 0) {
+    output.error(
+      `Invalid number of arguments. Usage: ${chalk.cyan(
+        `${getCommandName('target ls')}`
+      )}`
+    );
+    return 2;
+  }
+
+  const start = Date.now();
+  const projectUrl = `https://vercel.com/${link.org.slug}/${link.project.name}`;
+  const projectSlugLink = output.link(
+    chalk.bold(`${link.org.slug}/${link.project.name}`),
+    projectUrl,
+    {
+      fallback: () => chalk.bold(`${link.org.slug}/${link.project.name}`),
+      color: false,
+    }
+  );
+
+  output.spinner(`Fetching custom environments for ${projectSlugLink}`);
+
+  let url = `/projects/${encodeURIComponent(
+    link.project.id
+  )}/custom-environments?limit=20`;
+
+  const next = argv['--next'] || false;
+  if (next) {
+    url += `&until=${next}`;
```

Comment:
Blocking: Make sure to test this when pagination is added.

Reply:
Since there will be hard limits on the amount of custom environments users can have, we are not implementing pagination yet. Removed this pagination code.